travertino/src/travertino/properties/aliased.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from warnings import filterwarnings, warn
+
+# Make sure deprecation warnings are shown by default
+filterwarnings("default", category=DeprecationWarning)
+
+
+class Condition:
+    def __init__(self, **properties):
+        """A condition describing the values of one or more properties of a style.
+
+        :param properties: Any number of name: value pairs
+        """
+        self.properties = properties
+
+    def match(self, style):
+        return all(style[name] == value for name, value in self.properties.items())
+
+    def __str__(self):
+        return "; ".join(
+            [f"{name} == {value}" for name, value in self.properties.items()]
```

Comment:
A really minor optimisation - there's no need to roll out the full list with a comprehension; you can use a generator with join.
```suggestion
            f"{name} == {value}" for name, value in self.properties.items()
```

Reply:
I was under the impression that, while it's not *necessary*, it is slightly faster, since `join` [always builds a list anyway](https://stackoverflow.com/a/37782238/12975140). I am having trouble finding a more recent claim, though — is that no longer the case?

Reply:
I believe this is the equivalent spot in the current code:

https://github.com/python/cpython/blob/6e5b9f3d00658ce07909d78d1a796e95ad5dba1a/Objects/stringlib/join.h#L25

Reply:
I think the small increase in readability would almost always outweigh the small decrease in speed.

Reply:
Fair enough. I don't think I find it any less legible, so I've been in the habit of giving `join` list comps as a general best practice. But I have no objection to the opposite as a style guide. It's not like the performance difference is likely to ever actually matter.

Reply:
Interesting... TIL that this isn't as straightforward as I thought. Agreed that in this case, the performance implication isn't really a consideration; as for readability... it's fairly marginal either way. Regardless, the code has landed now, so it's probably not worth updating.